tests/components/mqtt/test_legacy_vacuum.py

```diff
@@ -251,8 +213,6 @@
         "fan_speed": "max"
     }"""
     async_fire_mqtt_message(hass, 'vacuum/state', message)
-    await hass.async_block_till_done()
```

Comment:
Why don't we need to block till done here?

Reply:
Is it using `async_write_ha_state` directly?

Reply:
Yeah, https://github.com/home-assistant/home-assistant/blob/e3e7fb5ff61b217c1c629e610b105481882d8d22/homeassistant/components/mqtt/vacuum/schema_legacy.py#L297
It should be the same implementation for all MQTT platforms.

Reply:
Kudos to @MartinHjelmare 

I can't believe someone is actually reviewing such refactoring pull request.

Reply:
Yeah, I fully agree. A big thanks to @martinhjelmare!

Reply:
Good and thanks! :smile: